content/en/_index.md

```diff
@@ -148,6 +179,14 @@ public static void exampleSnippet(String projectId, String filePath) {
     // Snippet content ...
 }
 {{< /tab >}}
+{{< tab header="Go" >}}
+// exampleSnippet demonstrates best practices.
+// The function prints to an io.Writer for testing purposes. Do not print to
+// stdout or stderr. Use the io.Writer instead.
+func exampleSnippet(w io.Writer, projectId, filePath string) error {
+    // Snippet content ...
+}
```

Comment:
I think using an error return type here goes against the guideline:

> Methods should avoid a return type whenever possible. Instead, show the user how
to interact with a returned object programmatically by printing some example
attributes to the console.

Reply:
This might be worth updating, given that most other languages here (Java, Python, Node) will use exceptions. Granted, we could write an error to the io.Writer, but for Go that's pretty unusual.

Reply:
This is actually intentional, the goal is to force the sample to write what went wrong rather than rely on the test to catch errors. [Here's a similar example in Java](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/9f6a4731b112e3554bc245abb0836b81e5acaf7c/iam/api-client/src/main/java/iam/snippets/DisableServiceAccount.java#L56).

Reply:
So the idea is that both success and failure are reported to the console. I like the symmetry.

Reply:
I am not sure I agree with this for Go. We would be forced to log.Fatal in samples, which is a panic and anti-pattern. I understand the proposition for exception based languages, but I don't think this rule translates well for Go.

Reply:
Resolving this to consolidate the discussion on this thread: https://github.com/GoogleCloudPlatform/samples-style-guide/pull/43#pullrequestreview-917877148.